controllers/onload_controller.go

```diff
@@ -858,12 +859,19 @@ func createModule(
 
 const devicePluginNameSuffix = "-onload-device-plugin-ds"
 const onloadVersionLabel = "onload-version"
+const devicePluginImageEnv = "DEVICE_PLUGIN_IMG"
 
 func (r *OnloadReconciler) createDevicePluginDaemonSet(
 	ctx context.Context, onload *onloadv1alpha1.Onload,
 ) (*ctrl.Result, error) {
 	log := log.FromContext(ctx)
 
+	devicePluginImage := os.Getenv(devicePluginImageEnv)
+	if devicePluginImage == "" {
+		return nil, fmt.Errorf("%q environment variable empty or not set."+
```

Comment:
Maybe even refuse to start the controller if `DEVICE_PLUGIN_IMG` is not set?

One example is KMM in `main()`:
```go
workerImage := cmd.GetEnvOrFatalError("RELATED_IMAGES_WORKER", setupLogger)
```
(https://github.com/kubernetes-sigs/kernel-module-management/blob/940e25e11ed701e6c218577066e29cadb4d9a62d/cmd/manager/main.go#L87)

Reply:
Done as a new commit, if you're happy with it I'll squash it before merging the PR